File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/organisms/section/OrgTextSection.kt
package com.perrystreet.woof.designsystem.atomic.organisms.section

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.SpacingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.text.AtomText
import com.perrystreet.woof.designsystem.atomic.atoms.text.roles.TextFontRole
import com.perrystreet.woof.designsystem.atomic.molecules.header.MolSectionTitle
import com.perrystreet.woof.designsystem.atomic.molecules.header.roles.SectionToneRole

@Composable
fun OrgTextSection(
    title: String,
    text: String,
    toneRole: SectionToneRole = SectionToneRole.OnScrim,
) {
    Column(
        modifier = with(toneRole) { Modifier.fillMaxWidth().sectionContainer() },
        verticalArrangement = Arrangement.spacedBy(SpacingRoles.Component.Cozy.dp),
    ) {
        MolSectionTitle(title = title, toneRole = toneRole)
        AtomText(
            text = text,
            textFontRole = TextFontRole.BodyP1,
            colorRole = toneRole.bodyColorRole,
        )
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Organisms/Section/OrgTextSection.swift
import SwiftUI

public struct OrgTextSection: View {
    @Environment(\.theme) private var theme

    private let title: String
    private let text: String
    private let toneRole: SectionToneRole

    public init(title: String, text: String, toneRole: SectionToneRole = .onScrim) {
        self.title = title
        self.text = text
        self.toneRole = toneRole
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: SpacingRoles.Component.cozy.rawValue) {
            MolSectionTitle(title: title, toneRole: toneRole)
            AtomText(text: text, textFontRole: .bodyP1, colorRole: toneRole.bodyColorRole)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(toneRole.containerPadding(from: theme))
        .background(toneRole.containerColor(from: theme), in: RoundedRectangle(cornerRadius: theme.sizing.radiusL))
    }
}
